core/src/sigverify_stage.rs

```diff
@@ -266,6 +273,21 @@ impl SigVerifyStage {
         }
     }
 
+    fn maybe_shrink_batches(packet_batches: &mut Vec<PacketBatch>) -> (u64, usize) {
+        let mut shrink_time = Measure::start("sigverify_shrink_time");
+        let num_packets = count_packets_in_batches(packet_batches);
+        let num_discarded_packets = count_discarded_packets(packet_batches);
+        let pre_packet_batches_len = packet_batches.len();
+        let discarded_packet_rate = (num_discarded_packets as f64) / (num_packets as f64);
```

Comment:
f32 is fine here

Reply:
Also need to check `num_packets` isn't 0... I patched the v1.9 and v1.10 backports to avoid divide by zero but still need to patch master